### DISCUSSION TITLE
Is there any way to set slint ui object's scale or matrix?

### DISCUSSION THREAD

**Discussion by fluolite** [slint-ui/slint#6227] [General]
Hi, my respectable friend.
Is there any way to set slint ui object's scale or matrix?
I can't find it.

**Comment by ogoffart** [comment]
Not at the moment.
An Image or Text can be rotated with the rotation property. An image scaled by changing it's width and height.
But it is not possible to rotate or scale arbitrary elements.
What is your usecase?

(Generic rotation tracked in issue https://github.com/slint-ui/slint/issues/1481)

**Reply by felipetesc** [reply]
usecase: People with eye problems, such as myself.

**Reply by ogoffart** [reply]
For that usecase, there is the scale factor that can scale the whole UI (SLINT_SCALE_FACTOR env variable) as well as bigger default fonts.
It is also possible to rotate the whole UI with the software renderer to render on screen that is in portrait mode or some other orientation.

But I don't think that was the original use case.

**Reply by fluolite** [reply]
usecase: 
Sometimes, a TouchArea touch pressed, UI target scale as an interactive feedback！
When I change it's width/height, may affect others binding.

**Reply by fluolite** [reply]
I'm a little confused, scale Should not be used as a settable property for UI elements.

**Reply by ApolloRoboto** [reply]
Hi, I have a use case, I wanted to build a pan/zoom component to navigate a graph, having a scale property would help a lot if it scales itself and all it's children.
Also scaling a component as it is dragged by the user is a nice visual feedback to have.

Are there any plans on adding such a property on the base components?

**Reply by MovGP0** [reply]
I'm having the same problem. trying to implement zoom/pan/rotate, but without a global transformation matrix, i keep producing bugs, causing me to write additional tests, properties for measurement, and workarounds, which I would not have otherwise.
